bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/XmlFactory.java

```diff
@@ -80,8 +79,6 @@ public static BXml parse(String xmlStr) {
 
             XmlTreeBuilder treeBuilder = new XmlTreeBuilder(xmlStr);
             return treeBuilder.parse();
-        } catch (BError e) {
-            throw e;
         } catch (Throwable e) {
             throw ErrorCreator.createError(StringUtils.fromString(("failed to parse xml: " + e.getMessage())));
```

Comment:
Shall we print the stack trace if the error message is null?

Reply:
Addressed with d04c6a717e035d66e5b732057429e1e20d22be8b.

Reply:
Reverted the change as the lines were not covered.